### PR TITLE
A few mini tests for linalg

### PR DESCRIPTION
**Pull request by kshyatt**
`broadcast` methods for rounding to a specific type were tested. But rounding "in general" wasn't for `{Sym}Tridiagonal`s.

### CODE REVIEW

File: test/linalg/tridiag.jl [hunk 1]
```diff
@@ -50,9 +50,21 @@ for elty in (Float32, Float64, Complex64, Complex128, Int)
     @test ctranspose(T) == Tridiagonal(conj(du), conj(d), conj(dl))
 
     @test abs.(T) == Tridiagonal(abs.(dl),abs.(d),abs.(du))
+    if elty <: Real
+        @test round.(T) == Tridiagonal(round.(dl),round.(d),round.(du))
```
Comment (andreasnoack) [inline test/linalg/tridiag.jl]: Maybe also test the types here.

**Review by andreasnoack — APPROVED** [review]

File: test/linalg/tridiag.jl (hunk 1 above)
Comment (kshyatt) [inline test/linalg/tridiag.jl]: Good call. Would `@inferred` be what I want or just `isa`?

Comment (andreasnoack) [inline test/linalg/tridiag.jl]: I think `isa` because the most likely problem is that these suddenly end up as `Array`s.

**Comment by kshyatt** [comment]
I noticed that `cholesky.jl` didn't have the special `DimensionMismatch` throws for `RowVector`s so I added that and tests too...

**Comment by kshyatt** [comment]
@andreasnoack still ok?

File: base/linalg/cholesky.jl [hunk 2]
```diff
@@ -491,6 +491,21 @@ function A_ldiv_B!(C::CholeskyPivoted, B::StridedMatrix)
     end
 end
 
+\(::Cholesky, ::RowVector) = throw(DimensionMismatch("Cannot left-divide matrix by transposed vector"))
+\(::CholeskyPivoted, ::RowVector) = throw(DimensionMismatch("Cannot left-divide matrix by transposed vector"))
+\(::Cholesky{<:Number}, ::RowVector{<:Number}) = throw(DimensionMismatch("Cannot left-divide matrix by transposed vector"))
```
Comment (andreasnoack) [inline base/linalg/cholesky.jl]: Why is this necessary after the definition two lines above?

File: base/linalg/cholesky.jl [hunk 3]
```diff
@@ -491,6 +491,21 @@ function A_ldiv_B!(C::CholeskyPivoted, B::StridedMatrix)
...
+\(::CholeskyPivoted{<:Number}, ::RowVector{<:Number}) = throw(DimensionMismatch("Cannot left-divide matrix by transposed vector"))
+
+At_ldiv_B(::Cholesky, ::RowVector) = throw(DimensionMismatch("Cannot left-divide matrix by transposed vector"))
+At_ldiv_B(::Cholesky{<:Number}, ::RowVector{<:Number}) = throw(DimensionMismatch("Cannot left-divide matrix by transposed vector"))
```
Comment (andreasnoack) [inline base/linalg/cholesky.jl]: Same here

**Review by andreasnoack — APPROVED** [review]
LGTM. I'm only wondering about the almost repeated definitions.

File: base/linalg/cholesky.jl (hunk 2 above)
Comment (kshyatt) [inline base/linalg/cholesky.jl]: I was following what happens in `bidiagonal.jl`. cc @andyferris

File: test/linalg/bidiag.jl [hunk 4]
```diff
@@ -168,6 +174,14 @@ srand(1)
                 @test isa(round.(Int, T), Bidiagonal)
                 @test ceil.(Int,T) == Bidiagonal(ceil.(Int,T.dv), ceil.(Int,T.ev), T.isupper)
                 @test isa(ceil.(Int,T), Bidiagonal)
+                @test floor.(T) == Bidiagonal(floor.(T.dv), floor.(T.ev), T.isupper)
+                @test isa(floor.( T), Bidiagonal)
```
Comment (Sacha0) [inline test/linalg/bidiag.jl]: Extra space? (`( T)`)

Comment (kshyatt) [inline test/linalg/bidiag.jl]: nice catch <3

File: base/linalg/cholesky.jl (hunk 2 above)
Comment (andyferris) [inline base/linalg/cholesky.jl]: So it's been decided (for now at least) that we allow `vector*matrix` and `matrix*rowvector` if matrix is `n × 1` or `1 × n`, respectively (my original PR for rowvector forbade both of these, as well as the associated `\` and `/` methods). I suppose we should be consistent between `*` and `\` and `/` and remove these for cholesky and bidiagonal and everywhere else too. Unfortunately these here are a bit useless - they can only possibly work for `1×1` cholesky and length-1 rowvector (I even recently found out base supports 1-vector * 1-vector -> 1×n matrix!)

@stevengj - thoughts?

Comment (kshyatt) [inline base/linalg/cholesky.jl]: Well, we have tests for `Cholesky{Number}` for other things so if those can stay I think these can too.

Comment (andreasnoack) [inline base/linalg/cholesky.jl]: @andyferris My question was mainly about the need for both a method for `Cholesky` and one for `Cholesky{<:Number}`. Anyway, given the `*` decision, I think it makes sense just to remove these definitions and just throw `DimensionMismatch`.

Comment (andyferris) [inline base/linalg/cholesky.jl]: Yes - sorry I wasn't clear, I think these (and others) need to be deleted in order to be consistent. For `*`, `/` and `\`, just let rowvector behave like a matrix and let the existing code see if the dimensions mismatch (some methods might be needed to avoid ambiguities - in this case they need to reshape the rowvector into a matrix).

Comment (andyferris) [inline base/linalg/cholesky.jl]: Now I understand the original question (lines 3 and 4 here alarmed me much less than lines 1 and 2 did) - @kshyatt the reason for that in bidiagonal was purely to avoid ambiguity errors, so only define such methods where necessary (but that seems irrelevant if we are deleting all 4 methods).

File: base/linalg/cholesky.jl (hunk 3 above)
Comment (andyferris) [inline base/linalg/cholesky.jl]: We possibly need to remove all these methods to be consistent with #20423... sorry @kshyatt

Comment (andyferris) [inline base/linalg/cholesky.jl]: See https://github.com/JuliaLang/julia/issues/20389#issuecomment-277263812 for rationale.

Comment (kshyatt) [inline base/linalg/cholesky.jl]: @andyferris please *don't* apologize. This is why I pinged you! I'm glad we found out *now*, rather than after the PR had been merged. Thanks for explaining and pointing this out.

Comment (kshyatt) [inline base/linalg/cholesky.jl]: Update: @andyferris if I delete *all* the methods, then there's a `MethodError` in the tests... should we define a generic freakout `DimensionMismatch` for `Factorization`s?

Comment (andreasnoack) [inline base/linalg/cholesky.jl]: Isn't it just that `At_ldiv_B!` is not defined for `Cholesky`? In that case I'd just remove the test.

**Comment by kshyatt** [comment]
CI passed and I think I addressed everything?
